test/units/parsing/test_ajson.py

```diff
@@ -20,3 +25,138 @@ def test_AnsibleJSONDecoder_vault():
+
+
+def vault_data():
+    """
+    Prepares data for testing handling AnsibleVaultEncryptedUnicode
+    in AnsibleJSONEncoder.default()
+    """
+
+    with open(os.path.join(os.path.dirname(__file__), 'fixtures/ajson.json')) as f:
+        data = json.load(f, cls=AnsibleJSONDecoder)
+
+    data_0 = data['password']
+    data_1 = data['bar']['baz'][0]['password']
+
+    expected_0 = (u'$ANSIBLE_VAULT;1.1;AES256\n34646264306632313333393636316'
+                  '562356435376162633631326264383934326565333633366238\n3863'
+                  '373264326461623132613931346165636465346337310a32643431383'
+                  '0316337393263616439\n646539373134633963666338613632666334'
+                  '65663730303633323534363331316164623237363831\n35363335613'
+                  '93238370a313330316263373938326162386433313336613532653538'
+                  '376662306435\n3339\n')
+
+    expected_1 = (u'$ANSIBLE_VAULT;1.1;AES256\n34646264306632313333393636316'
+                  '562356435376162633631326264383934326565333633366238\n3863'
+                  '373264326461623132613931346165636465346337310a32643431383'
+                  '0316337393263616439\n646539373134633963666338613632666334'
+                  '65663730303633323534363331316164623237363831\n35363335613'
+                  '93238370a313330316263373938326162386433313336613532653538'
+                  '376662306435\n3338\n')
+
+    return [
+        (data_0, expected_0),
+        (data_1, expected_1),
+    ]
+
+
+class TestAnsibleJSONEncoder:
+    """
+    Namespace for testing AnsibleJSONEncoder
+    """
+
+    @pytest.fixture(scope='class')
+    def mapping(self, request):
+        """
+        Returns object of Mapping mock class.
+        The object is used for testing handling of Mapping objects
+        in AnsibleJSONEncoder.default().
+        Using a plain dictionary instead is not suitable because
+        it is handled by default encoder of the superclass (json.JSONEncoder).
+        """
+        class M(Mapping):
+            def __init__(self, *args, **kwargs):
+                self.__dict__.update(*args, **kwargs)
+
+            def __getitem__(self, key):
+                return self.__dict__[key]
+
+            def __iter__(self):
+                return iter(self.__dict__)
+
+            def __len__(self):
+                return len(self.__dict__)
+
+        return M(request.param)
+
+    @pytest.fixture
+    def ansible_json_encoder(self):
+        return AnsibleJSONEncoder()
+
+    ###############
+    # Test methods:
+
+    @pytest.mark.parametrize(
+        'test_input,expected',
+        [
+            (datetime(2019, 5, 14, 13, 39, 38, 569047), '2019-05-14T13:39:38.569047'),
+            (datetime(2019, 5, 14, 13, 47, 16, 923866), '2019-05-14T13:47:16.923866'),
+            (date(2019, 5, 14), '2019-05-14'),
+            (date(2020, 5, 14), '2020-05-14'),
+            (datetime(2019, 6, 15, 14, 45, tzinfo=tz("UTC")), '2019-06-15T14:45:00+00:00'),
```

Comment:
Could you please keep the same string quoting style across the module?
```suggestion
            (datetime(2019, 6, 15, 14, 45, tzinfo=tz('UTC')), '2019-06-15T14:45:00+00:00'),
```